File: spring-boot-properties-kotlin/src/main/java/de/larmic/properties/PropertiesApplication.java
package de.larmic.properties;

import de.larmic.properties.service.HelloProperties;
import de.larmic.properties.service.HelloService;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.context.properties.EnableConfigurationProperties;

@SpringBootApplication
@EnableConfigurationProperties(value = HelloProperties.class)
public class PropertiesApplication {

    public static void main(String[] args) {
        final var app = SpringApplication.run(PropertiesApplication.class, args);
        final var helloService = app.getBean(HelloService.class);
        helloService.logHelloProperties();
    }

}

File: spring-boot-properties-kotlin/src/main/kotlin/de/larmic/properties/PropertiesApplication.kt
package de.larmic.properties

import de.larmic.properties.service.HelloProperties
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.boot.runApplication

@SpringBootApplication
@EnableConfigurationProperties(value = [HelloProperties::class])
open class PropertiesApplication

fun main(args: Array<String>) {
    runApplication<PropertiesApplication>(*args)
}